plugin/core/rpc.py

```diff
@@ -118,39 +185,84 @@ def send_request(
             handler: Callable[[Optional[Any]], None],
             error_handler: Optional[Callable[[Any], None]] = None,
     ) -> None:
-        self.request_id += 1
         if self.transport is not None:
-            self.logger.outgoing_request(self.request_id, request.method, request.params, blocking=False)
-            self._response_handlers[self.request_id] = (handler, error_handler)
-            self.send_payload(request.to_payload(self.request_id))
+            with self._sync_request_cvar:
+                self.request_id += 1
+                self.logger.outgoing_request(self.request_id, request.method, request.params, blocking=False)
+                self._response_handlers[self.request_id] = (handler, error_handler)
+                self.send_payload(request.to_payload(self.request_id))
         else:
             debug('unable to send', request.method)
             if error_handler is not None:
                 error_handler(None)
             return None
 
-    def execute_request(self, request: Request, timeout: float = DEFAULT_SYNC_REQUEST_TIMEOUT) -> Optional[Any]:
+    def execute_request(
+            self,
+            request: Request,
+            handler: Callable[[Optional[Any]], None],
+            error_handler: Optional[Callable[[Any], None]] = None,
```

Comment:
What's the reason for this API to change from returning a value to passing through a callback?
I think that for a sync API it would make more sense to return a value (could be a tuple of (result, error) like in  some subprocess APIs).

Reply:
After looking at it a bit more, I suppose it's due to wanting to print the error when error_handler is not provided?